django/openQuestion.txt

```diff
@@ -0,0 +1,85 @@
+1. 
+    file typically defines the data models used in the application
+    part of the backend responsible for defining the database schema and handling data-related operations.
+2. 
+    This file is responsible for defining the views or endpoints that handle incoming HTTP requests
+    part of the backend, as it determines how the application responds to client requests
+3.
+    A file that holds list of urls that our project support. this list, map the url to view function that will handle 
+    the client requests
+    urls.py maps specific URL patterns to the corresponding Django application's urls.py file to handle them.
+4. 
+    To manage and track changes to the database schema over time. 
+    They are essential for database schema version control and data schema updates.
+    makemigrations: This command generates new migration files based on the changes you made to your models.
+    migrate: This command applies the pending migrations to update the database schema.
+5. 
+    The role of a serializer is to convert complex data types, such as Django models, into JSON or other data formats,
+    making it easy to render them into a content type suitable for client consumption
+6.
+    A toolkit built on top of the Django web framework that reduces the amount of code you need to write to create REST interfaces
+7.
+    serializers are custom classes used to transform complex data types like models into native data types
+    model serializers its a shortcut for creating serializers for model instances.
+    it provide simple way to create serializer classes for models.
+8.
+    1- One-To-One Relationship
+    2- One-To-Many Relationship
```

Comment:
Worth noting that it's defined with `ForeignKey`, and that it can be used to create Many-To-One relationship too.